src/hotrod/impl/protocol/HeaderParams.cpp

```diff
@@ -82,6 +82,10 @@ uint8_t HeaderParams::toOpRespCode(uint8_t code) {
         return HotRodConstants::EXEC_RESPONSE;
     case HotRodConstants::QUERY_REQUEST:
         return HotRodConstants::QUERY_RESPONSE;
+    case HotRodConstants::AUTH_MECH_LIST_REQUEST:
+        return HotRodConstants::AUTH_MECH_LIST_RESPONSE;
+    case HotRodConstants::AUTH_REQUEST:
+        return HotRodConstants::AUTH_RESPONSE;
```

Comment:
I've tried exactly this change in https://issues.jboss.org/browse/HRCPP-373. The point here is that all the HotRodConstants:: are defined (not just declared) in the class definition. This set some constraints on how to use these constants: i.e. is fine to use them in a switch but not in a map (you got an 'unresolved symbol'). So this change requires the move of all the constants defs in a .cpp file. I didn't go too deep in the details (https://gcc.gnu.org/wiki/VerboseDiagnostics#missing_static_const_definition), just considered it doesn't worth the effort: even if switch it's not cool it's still easy to read.